### DISCUSSION TITLE
PrimeFaces Ajax Reset Inputs Not Working for ViewScoped Bean

### DISCUSSION THREAD

**Discussion by TheMightyE** [primefaces/community#1877] [PrimeFaces] [answered]
By backing bean is ViewScoped. Within a dialog, I have one field, a Clear and an Add button. The Clear button performs an ajax reset input. If I enter a new value in the 'Title' field and click Add, I am unable to reset this field with my Clear button.

**To reproduce**
1. Extract the provided files to load up your project and go to localhost:8080/manageLibrary.xhtml.
2. Click the New button on top right to add a new item.
3. Enter any value in the Title field.
4. Click Add to submit the form.
5. Click Clear. You will notice the field still has the previously entered value and is not cleared.

What is the gap in my understanding of scopes and how can I clear the fields correctly?

PrimeFaces v. 13.0.8

[manageLibrary.zip](https://github.com/primefaces/community/files/15279628/manageLibrary.zip)

**Comment by melloware** [comment, marked as answer]
Fixed reproducer: 
[discussion-1877.zip](https://github.com/primefaces/community/files/15282864/discussion-1877.zip)


@TheMightyE it is your misunderstanding.  If you look at the Showcase: https://www.primefaces.org/showcase/ui/misc/resetInput.xhtml?jfwid=53569 the beans are `@RequestScoped` so they are cleared on every request what is being `reset` is the state of the JSF tree components for ERROR state etc.

Because you are using `@ViewScoped` it is resetting the value and state when you press clear however its resetting it to the value stored in your backing bean which is `@ViewScoped` and has a value where `@RequestScoped` does not.

so if you want this behavior with a ViewScoped bean I suggest your Clear button add ..

```xml
<p:ajax update="panel" resetValues="true" listener="#{libraryBean.reset}" />
```

```java
	public void reset() {
		enteredTitle = "";
	}
```

**Reply by TheMightyE** [reply]
Thank you!

**Reply by TheMightyE** [reply]
Is there a way to reset the inputs from the backing bean as well. I want to reset the fields after a successful add operation. I modified my add function to call `reset()` and `PrimeFaces.current().resetInputs("dialogForm:addDialog")` at the end but this did not reset the field.

**Modified `addItem()` function**
```java
    public void addItem() {
        if (enteredTitle.isBlank()) {
            FacesContext.getCurrentInstance().addMessage(null,
                    new FacesMessage(FacesMessage.SEVERITY_ERROR, "Missing Field.", "Title is required"));
            return;
        }
        if (enteredTitle.length() < 2) {
            FacesContext.getCurrentInstance().addMessage(null,
                    new FacesMessage(FacesMessage.SEVERITY_ERROR, "Invalid Title.", "Title must be greater than 2 characters."));
            return;
        }
        switch (selectedLibrary) {
            case "Movies":
                movies.add(enteredTitle);
                break;
            case "Books":
                books.add(enteredTitle);
                break;
            default:
                break;
        }
        reset();
        PrimeFaces.current().resetInputs("dialogForm:addDialog");
    }
```

**Reply by melloware** [reply]
Hmmm your code should work fine.  The `enteredTitle` should be blank now

**Reply by TheMightyE** [reply]
Okay I see. It's okay if I cannot get it to work. Your solution for the Clear button is good enough for now.
